src/TuxAnimations.js

```diff
@@ -45,75 +46,131 @@ var makeAnimation = function (transitions, customClassName) {
-            //Case to accept if delay prop is a number
-            if (typeof delay === "number") {
-              delay = delay + "ms";
+            //Delay prop work--------------------------------------------
+            var delay = this.props.delay;
+            //Use default if no delay prop defined
+            if (delay) {
+              //Case to accept if delay prop is a number
+              if (typeof delay === "number") {
+                delay = delay + "ms";
+              }
+              transitions[action]['transition-delay'] =  delay;
             }
-            transitions[action]['transition-delay'] =  delay;
-          }
 
-          //Custom properties work-------------------------------------
-          var custom = this.props.custom;
-          if (custom) {
-            for (var cssElement in custom) {
-              transitions['enterActive'][cssElement] =  custom[cssElement];
-              transitions['leave'][cssElement] =  custom[cssElement];
+            //Custom prop work-------------------------------------
+            var custom = this.props.custom;
+            if (custom) {
+              for (var cssElement in custom) {
+                transitions['enter-active'][cssElement] =  custom[cssElement];
+                transitions['leave'][cssElement] =  custom[cssElement];
+              }
             }
-          }
-        }
+          }//End for css in transitions[action]
+        }//End if action === className
       }//End for loop for transitions
+    },
 
+    componentWillEnter: function (callback) {
+      this.setAnimationDomNode('enter', callback);
+    },
+
+    componentWillLeave: function (callback) {
+      this.setAnimationDomNode('leave', callback);
+    },
+
+    render: function () {
       //Return new React.Dom element
       return (
         React.DOM.div(
           {
             className: className
           },
-           this.props.children
+          this.props.animate
         )
       );
     }
    });
 };
 
-module.exports = makeAnimation;
+// createAnimationGroup FUNCTION: wraps custom animation component in a ReactTransitionGroup
+  // @param Animation OBJECT: animation component based on custom properties
+var createAnimationGroup = function (Animation, customClassName, tagToRender) {
+
+  tagToRender = tagToRender || 'span';
+
+  //React.createClass FUNCTION: function to create animation component
+    //@param OBJECT: component setState, props, and render
+  return React.createClass({
+    // Set toAnimate initial state as an array that wraps custom Animation component
+    // *** toAnimate is an array because ReactTransitionGroup only accepts multiple elements if wrapped in a single array
+    getInitialState: function () {
+      return {
+        toAnimate: []
+      }
+    },
+    //Used to update toAnimate
+    componentWillReceiveProps: function (newProps) {
+      this.setState({
+        toAnimate: this.state.toAnimate.concat(newProps.children)
```

Comment:
I still believe that this needs to be changed to `toAnimate: [].concat(newProps.children)`.  Think about it this way.  Lets say we add a list of rooms to animate.  So they will be in our `toAnimate` array.  Then lets say we delete one of the rooms.  That means we are gonna submit new props, which is going to be the rooms array minus the one room we just deleted.  However, because we are concating this list onto the existing `toAnimate` array, the room will still be in the array from the first time `componentWillReceiveProps` was invoked.  Try testing animating the messages in a room and try adding and then removing messages to verify but I believe we will get an issue in that case.